crates/llm-ls/src/retrieval.rs

```diff
@@ -414,8 +431,12 @@ impl SnippetRetriever {
                     1,
                     TruncationDirection::Left,
                 );
-                self.generate_embedding(encoding, self.model.clone()).await
+                self.generate_embeddings(vec![encoding], self.model.clone()).await?
             }
+        };
+        match result.first() {
```

Comment:
I'd rather have an `.is_empty` check and then call `Vec::remove`. This we way we avoid cloning the value needlessly and since the `Vec` should contain only one value I don't think the `remove` call has any overhead